In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from scipy import stats
from grab_db import my_db
from pydataset import data

#### For each of the following questions, formulate a null and alternative hypothesis (be as specific as you can be), then give an example of what a true positive, true negative, type I and type II errors would look like. Note that some of the questions are intentionally phrased in a vague way. It is your job to reword these as more precise questions that could be tested.

### 1 Has the network latency gone up since we switched internet service providers?

Hnull - Network latency is the same between both internet service providers. <br>
Halt - The network latency has gone up since we switched internet service providers.

TP : we concluded network latency went up and it did <br>
TN: We concluded network latency did not go up and it did not

type 1 error: We reject the null but it is true <br>
type 2 error: we failed to reject the null when it was not true


### 2 Is the website redesign any good?

Hnull - the website design has not increased clicks. <br>
Halt - the website design is better now and has increased clicks.

TP: we concluded website design is better now and it is <br>
TN: We conluded website design is not better and it isnt

type 1 error: we reject the null but it is true <br>
type 2 error: we failed to reject the null when it was true

### 3 Is our television ad driving more sales?

Hnull - the television ad has had no impact on driving more sales. <br>
Halt - The television ad has increased sales.

TP: we concluded the ad increased sales and it did <br>
TN:we concluded ad had no impact on sales and it did not

type 1 error: we reject the null but it is true <br>
type 2 error: we failed to reject the null when it was true.

## T-TEST:

#### Answer with the type of test you would use (assume normal distribution):

##### Is there a difference in grades of students on the second floor compared to grades of all students?

- 1 sample T-test

##### Are adults who drink milk taller than adults who dont drink milk?

- independent 1 tail 2 sample T-test

##### Is the the price of gas higher in texas or in new mexico?

- independent 1 tail 2 sample T-test

##### Are there differences in stress levels between students who take data science vs students who take web development vs students who take cloud academy?


 - Anova test

##### Ace Realty wants to determine whether the average time it takes to sell homes is different for its two offices. A sample of 40 sales from office #1 revealed a mean of 90 days and a standard deviation of 15 days. A sample of 50 sales from office #2 revealed a mean of 100 days and a standard deviation of 20 days. Use a .05 level of significance.

 - 2 sample T-test
 - Hnull - average time to sell homes is the same between office 1 and 2 <br>
 - Halt - average time to sell homes is different between office 1 and 2

In [3]:
of_1 = np.random.normal(90,15,40)
of_2 = np.random.normal(100,20,50)
t, p = stats.ttest_ind(of_1, of_2)

TEST VARIANCE:

In [4]:
stat, p_val = stats.levene(of_1, of_2)

In [5]:
if p_val < .05:
    print('reject null')
else:
    print('we fail to reject the null')

reject null


according to levene test variance is equal.

In [6]:
t, p = stats.ttest_ind(of_1, of_2)

In [7]:
if p < .05:
    print('reject null')
else:
    print('We fail to reject the null')

reject null


This indicates that the average time to sell homes is the same between office 1 and office 2

#### Load the mpg dataset and use it to answer the following questions:

In [8]:
mpg = data('mpg')

##### Is there a difference in fuel-efficiency in cars from 2008 vs 1999?

 - 2 sample 2 tail T-test <br>
 - Hnull - there is no difference in fuel efficiency in cars from 2008 and 1999
 - Halt - there is a difference in fuel efficiency in cars from 2008 and 1999

In [9]:
mpg['FE'] =(mpg.cty + mpg.hwy)/2

In [10]:
fe_2008 = mpg.FE[mpg.year == 2008]

In [11]:
fe_1999 = mpg.FE[mpg.year == 1999]

TEST VARIANCE

In [12]:
stat, p_val = stats.levene(fe_2008, fe_1999)

In [13]:
if p_val < .05:
    print('we reject the null')
else:
    print('we fail to reject null')
    

we fail to reject null


According to levene test variance is equal.

In [14]:
t, p = stats.ttest_ind(fe_1999,fe_2008)

In [15]:
if p < .05:
    print('we reject the null')
else:
    print('we fail to reject null')
    

we fail to reject null


Since we failed to reject null we can conclude that there is no significant difference in fuel efficiency between 1999 and 2008 vehicles.

#### Are compact cars more fuel-efficient than the average car?

 - 1 sample ttest <br>
 - Hnull - compact cars are equally efficient to the average car.<br>
 - Halt - compact cars have a different efficiency than the average car.

In [34]:
compact_FE = mpg.FE[mpg['class'] == 'compact']
FE = np.mean(mpg.FE)

In [35]:
t, p = stats.ttest_1samp(compact_FE,FE)

In [36]:
if p < .05:
    print('we reject the null')
else:
    print('we fail to reject the null')

we reject the null


Since we have rejected the null we can coclude that there is a DIFFERENCE between avg efficiency and efficiency of compact cars. now we need to see if the compact cars are MORE efficient by observing their means.

In [38]:
compact_FE.mean()

24.21276595744681

In [39]:
FE

20.14957264957265

as you can see the compact cars have a mean fuel efficiency of 24.2 while the overall average is 20.1

#### Do manual cars get better gas mileage than automatic cars?

- 2 sample Ttest.
- Hnull - Manual cars get equal mileage to automatic cars <br>
- Halt - manual cars get better mileage than automatic cars.

In [67]:
Auto_FE = mpg.FE[mpg.trans.str[0] == 'a']
Manual_FE = mpg.FE[mpg.trans.str[0] != 'a']

TEST VARIANCE

In [71]:
stat, p_val = stats.levene(Auto_FE,Manual_FE)

In [72]:
if p_val < .05:
    print('we reject null')
else:
    print('we fail to reject null')

we fail to reject null


According to levene test variance is equal.

In [73]:
t, p = stats.ttest_ind(Manual_FE,Auto_FE)

In [74]:
if p < .05:
    print('we reject null')
else:
    print('we fail to reject null')

we reject null


Based on Ttest we can conclude there is a difference between manual and auto fuel efficiency. now we need to compare means to see if manual cars FE is better or worse.

In [75]:
Auto_FE.mean()

19.130573248407643

In [76]:
Manual_FE.mean()

22.227272727272727

Manual FE is better than Automatic FE.